include/dlaf/util_matrix.h

```diff
@@ -21,149 +21,111 @@ constexpr double M_PI = 3.141592;
 #include <blas.hh>
 #include <hpx/hpx.hpp>
 
+#include "dlaf/common/assert.h"
 #include "dlaf/common/index2d.h"
 #include "dlaf/matrix.h"
 #include "dlaf/types.h"
 
 /// @file
 
 namespace dlaf {
-namespace util_matrix {
+namespace matrix {
+namespace util {
+namespace internal {
 
 /// @brief Verify if dlaf::Matrix is square
 ///
 /// @tparam Matrix refers to a dlaf::Matrix object
```

Comment:
Documentation has to be improved.
- failure conditions are missing
- parameters documentation has to be fixed
- what happens in the case that one of the condition is not met.